Revert change causing regression in DAQ/HLT JSON input parsing (11X backport)

smorovic opened the pull request:
## PR description:
Reverts json_reader.cpp modification which was part of this commit:
d030f66

Fixes a bug that currently prevents 11_1_X release line from running in DAQ/HLT.

More information and description in #30315 of which this PR is a backport.

smorovic committed:
revert commit which resulted in regression in JSON parsing of DAQ input
source:
https://github.com/cms-sw/cmssw/commit/d030f663df51846c06f626488b8dfa2490d187bd

cmsbuild commented:
A new Pull Request was created by  @smorovic (Srecko Morovic) for CMSSW_11_1_X.

It involves the following packages:

EventFilter/Utilities


@perrotta, @mommsen, @cmsbuild, @emeschi, @slava77 can you please review it and eventually sign? Thanks.
@Martin-Grunewald this is something you requested to watch as well.
@silviodonato, @dpiparo you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

smorovic committed:
on suggestion that original condition is always false, it can simply be
removed

cmsbuild commented:
Pull request #30316 was updated. @perrotta, @mommsen, @cmsbuild, @emeschi, @slava77 can you please check and sign again.

perrotta commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 19b8572873ccbb77fc1431342eda15966b8f938d
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-08bc74/7238/summary.html
CMSSW: CMSSW_11_1_X_2020-06-21-2300
SCRAM_ARCH: slc7_amd64_gcc820

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-08bc74/7238/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 36
 - DQMHistoTests: Total histograms compared: 2780792
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2780741
 - DQMHistoTests: Total skipped: 50
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 35 files compared)
 - Checked 152 log files, 16 edm output root files, 36 DQM output files

smorovic committed:
Apply jsoncpp upstream fix:
https://github.com/open-source-parsers/jsoncpp/commit/9c98f2277b2972ddee93c97bb5273e86d6045195

cmsbuild commented:
Pull request #30316 was updated. @perrotta, @mommsen, @cmsbuild, @emeschi, @slava77 can you please check and sign again.

perrotta commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: aae3f9bc6ac90200102d33762bc649c23e66fb2a
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-08bc74/7246/summary.html
CMSSW: CMSSW_11_1_X_2020-06-22-1100
SCRAM_ARCH: slc7_amd64_gcc820

cmsbuild commented:
Comparison job queued.

emeschi commented:
+1

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-08bc74/7246/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 36
 - DQMHistoTests: Total histograms compared: 2780792
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2780741
 - DQMHistoTests: Total skipped: 50
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 35 files compared)
 - Checked 152 log files, 16 edm output root files, 36 DQM output files

perrotta commented:
+1
- Fixed a wrong expression implemented with #29592: original intention behind the code is now restored
- Jenkinns tests pass

perrotta commented:
backport of #30315

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_11_1_X IBs (tests are also fine) and once validation in the development release cycle CMSSW_11_2_X is complete. This pull request will now be reviewed by the release team before it's merged. @silviodonato, @dpiparo (and backports should be raised in the release meeting by the corresponding L2)

silviodonato commented:
+1
